pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -268,7 +279,63 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 		volumeName = util.GetUniqueVolumeNameFromSpecWithPod(podName, volumePlugin, volumeSpec)
 	}
 
-	if _, volumeExists := dsw.volumesToMount[volumeName]; !volumeExists {
+	var seLinuxFileLabel string
+	// Volume plugin supports SELinux context mount for all its volumes.
+	var pluginSupportsSELinuxContextMount bool
+	// The volume is ReadWriteOncePod. We don't support other volume types in SELinuxMountReadWriteOncePod feature.
+	// Don't use mount option to apply the SELinux context, still, track the context and report metrics of things
+	// that would break if the feature was for all volume access modes.
+	var isRWOP bool
+
+	if feature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) {
+		pluginSupportsSELinuxContextMount, err = dsw.getSELinuxMountSupport(volumeSpec)
+		if err != nil {
+			return "", err
+		}
+		isRWOP = util.IsRWOP(volumeSpec)
+		if pluginSupportsSELinuxContextMount {
+			// Ensure that a volume that can be mounted with "-o context=XYZ" is
+			// used only by containers with the same SELinux contexts.
+			for _, containerContext := range seLinuxContainerContexts {
+				newLabel, err := util.SELinuxOptionsToFileLabel(containerContext)
+				if err != nil {
+					fullErr := fmt.Errorf("failed to construct SELinux label from context %q: %s", containerContext, err)
+					if isRWOP {
```

Comment:
This error processing is repeated a few times. Maybe worth putting into a function, especially if you plan to add metrics later

Reply:
Extracted into a function.